robottelo/decorators/__init__.py

```diff
@@ -239,11 +252,13 @@ def wrapper(*args, **kwargs):
             else:
                 settings_project = 'sat'
 
-            if project not in allowed_project_modes:
-                raise ProjectModeError(
-                    '"{0}" is not a project mode. The allowed project modes '
-                    'are: {1}'.format(project, allowed_project_modes)
-                )
+            for project in projects:
```

Comment:
allowed and project could be sets so no complex logic would be needed. Also you could return all wrong parameters at once. On current solution it will stop on the first wrong project. Ex:

```python
>>> allowed=set('sat sam upgrade'.split())
>>> projects={'sam'}
>>> projects-allowed
set()
>>> projects=set('foo bar sam'.split())
>>> projects-allowed
{'bar', 'foo'}
```
Current code will work but I think sharing the use of set for this kind task would be nice ;)

Reply:
Yeah ... using set can be effective. I will try to incorporate this to reduce the code and make code effective. 

But the question is whether we really want to get this PR merge. Becuase we have introduced new decorator for upgrade.

If people want to this I can keep else discard.